staging/src/k8s.io/apiserver/pkg/admission/chain.go

```diff
@@ -16,22 +16,46 @@ limitations under the License.
 
 package admission
 
-// chainAdmissionHandler is an instance of admission.Interface that performs admission control using a chain of admission handlers
-type chainAdmissionHandler []Interface
+import "time"
+
+// chainAdmissionHandler is an instance of admission.NamedHandler that performs admission control using
+// a chain of admission handlers
+type chainAdmissionHandler []NamedHandler
```

Comment:
@jpbetz This type is a union type of `Interfaces` to produce a union `Interface`.  This change breaks the composition model of having a union type that unions the types it is exposing.  Please make a followup that restores the composition model.

You can do this by creating an `Interface` wrapper which provides both `Admit` and `Validate`, and does type casting on the delegate.  That will preserve the abstractions and allows layering.